axonserver/src/main/java/io/axoniq/axonserver/grpc/HeartbeatMonitor.java

```diff
@@ -0,0 +1,143 @@
+import io.axoniq.axonserver.applicationevents.TopologyEvents.ApplicationInactivityTimeout;
+import io.axoniq.axonserver.grpc.control.Heartbeat;
+import io.axoniq.axonserver.grpc.control.PlatformInboundInstruction;
+import io.axoniq.axonserver.grpc.control.PlatformOutboundInstruction;
+import io.axoniq.axonserver.message.ClientIdentification;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.event.EventListener;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+import static io.axoniq.axonserver.grpc.control.PlatformInboundInstruction.RequestCase.HEARTBEAT;
+
+/**
+ * Responsible for sending and receiving the heartbeat signals to and from clients.
+ * Uses heartbeats to verify if the connections with clients are still alive.
+ *
+ * @author Sara Pellegrini
+ * @since 4.2.2
+ */
+@Component
+public class HeartbeatMonitor {
+
+    private final Clock clock;
+
+    private final long heartbeatTimeout;
+
+    private final ApplicationEventPublisher eventPublisher;
+
+    private final Publisher<PlatformOutboundInstruction> heartbeatPublisher;
+
+    private final Map<ClientIdentification, Instant> lastReceivedHeartBeats = new ConcurrentHashMap<>();
+
+    private final Map<ClientIdentification, String> clientComponents = new ConcurrentHashMap<>();
+
+
+    /**
+     * Constructs a {@link HeartbeatMonitor} that uses {@link PlatformService} to send and receive heartbeats messages.
+     *
+     * @param platformService  the platform service
+     * @param eventPublisher   the internal event publisher
+     * @param heartbeatTimeout the max period of inactivity before is published an {@link ApplicationInactivityTimeout}
+     */
+    @Autowired
+    public HeartbeatMonitor(PlatformService platformService,
+                            ApplicationEventPublisher eventPublisher,
+                            @Value("${axoniq.axonserver.client-heartbeat-timeout:5000}") long heartbeatTimeout) {
+        this(listener ->
+                     platformService.onInboundInstruction(HEARTBEAT, (client, context, instruction) ->
+                             listener.accept(new ClientIdentification(context, client), instruction)),
+             eventPublisher,
+             platformService::sendToAllClient,
+             heartbeatTimeout, Clock.systemUTC());
+    }
+
+    /**
+     * Primary constructor of {@link HeartbeatMonitor}
+     *
+     * @param heartbeatListenerRegistration consumers of heartbeats listener
```

Comment:
More so a personal flavor thing, but I typically like to not only describe the parameter, but also describe what it's used for. Just a thought, not a harsh requirement from my part.